packages/migrate/src/utils/getDatabaseVersionSafe.ts

```diff
@@ -0,0 +1,28 @@
+import Debug from '@prisma/debug'
+
+import { MigrateEngine } from '../MigrateEngine'
+
+const debug = Debug('prisma:cli')
+
+/**
+ * Retrieve the database version from the given schema or url.
+ * This function never throws, and was introduced to prevent circular dependencies in `@prisma/internals`.
+ */
+export async function getDatabaseVersionSafe(schemaOrUrl: string): Promise<string | undefined> {
+  let engine: MigrateEngine | undefined
+  let dbVersion: string | undefined
+  try {
+    engine = new MigrateEngine({
+      projectDir: process.cwd(),
+    })
+    dbVersion = await engine.getDatabaseVersion({ schema: schemaOrUrl })
```

Comment:
Does this always start up a new Engine to make this query, or would it reuse an existing one if it already exists?